Project Auth by Max and Sandrine

MindstormingAB opened the pull request:
Link to our deployed project: https://max-sandrine-auth-app.netlify.app/
Link to our API: https://max-sandrine-auth-api.herokuapp.com/

Our reflections on how this week's project turned out: This week's project was pretty difficult because it required the new backend knowledge related to authentication we learned during the week as well as all the React/Redux skills we previously acquired during the Bootcamp. It was very challenging to build both frontend and backend from scratch.

Things I'd like to have clarified or explained in more detail: -

pwangy commented on frontend/src/index.css:
```diff
@@ -5,9 +5,99 @@ body {
+  color: white;
+  background-image: linear-gradient(#335b71 45%, #03324c 55%);
+  transition: color 0.3s, background-image 0.5s, ease-in-out;
+  /* border-radius: 25px; */
```
Remove commented out line.

pwangy commented on frontend/src/index.css:
```diff
@@ -5,9 +5,99 @@ body {
+}
+
+p {
+  margin: 0 0 16px 0;
```
perhaps just margin-bottom?

pwangy reviewed (approved):
Really nice work, you two. 💪 

This week's assignment was really difficult-- and you gave us a few reminders of things we had forgotten, so thank you for that! How nice that you went the extra step to show a personalized welcome screen upon login. All those little touches you added were appreciated. 

Most of the actionable comments are minor clean-up or consistency notes; or general food-for-thought. 😁

KarolCislowski commented on backend/server.js:
```diff
@@ -1,29 +1,147 @@
+  }
+});
+
+userSchema.pre("save", async function (next) {
```
nice use of  `pre()`

KarolCislowski commented on backend/server.js:
```diff
@@ -1,29 +1,147 @@
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
+const listEndpoints = require("express-list-endpoints");
```
You should import it on top of your script with `import/from` syntax.

KarolCislowski commented on backend/server.js:
```diff
@@ -1,29 +1,147 @@
+  origin: function (origin, callback) {
+    if (!origin) return callback(null, true);
+    if (allowedDomains.indexOf(origin) === -1) {
+      var msg = `This site ${origin} does not have an access. Only specific domains are allowed to access it`
```
Nice that you implement this logic, but remember that you shouldn't use `var` any more. `const` would be proper here. 🤓

KarolCislowski commented on backend/server.js:
```diff
@@ -1,29 +1,147 @@
+
+// Update endpoint
+app.post("/users/:userId/profile", authenticateUser);
+app.post("/users/:userId/profile", async (req, res) => {
```
It is not a good practice to pass any user data within a URL. You should use the request body to pass it to the server. 🤓

KarolCislowski commented on frontend/src/reducers/user.js:
```diff
@@ -0,0 +1,34 @@
+      state.login.userId = "";
+    },
+  }
+});
```
Reducer is a great place to manage your users but you need to store some data in the local storage to make them stay legged in after they refresh the page 🤓

KarolCislowski reviewed (changes requested):
It is a great app! Good job 🦄 

You need to fix only one issue with storing your data in the local storage.

Good luck 🌈

MindstormingAB commented:
Hi @KarolCislowski and @pwangy 
Thank you for reviewing this repository.
I did the following changes with regard to your feedback:
- added a local storage in frontend
- moved userId from the endpoint url to a header
- replaced POST with PATCH
- fixed smaller "bits and bobs"

KarolCislowski reviewed (approved):
Great work!
